pages/index.tsx

```diff
@@ -28,14 +29,46 @@ import {
   Synth,
 } from "../utils/umaApi";
 
-import { getDefillamaTvl, getDefillamaPercentChange } from "../utils/defillama";
+import {
+  getDefillamaTvl,
+  getDefillamaPercentChange,
+  getDefillamaStats,
+} from "../utils/defillama";
+import { ethers } from "ethers";
+
+async function attachDefillamaStats(cmsSynth: ContentfulSynth) {
+  let data;
+
+  try {
+    const { tvl, tvl24hChange } = await getDefillamaStats(cmsSynth.externalApi);
+    data = {
+      ...cmsSynth,
+      tvl: ethers.utils.parseEther(tvl.toFixed(0)).toString(),
+      tvl24hChange,
+    };
+  } catch (error) {
+    return error;
+  }
+
+  return data;
+}
+
+function fetchCompleteSynthByApi(cmsSynth: ContentfulSynth) {
+  if (cmsSynth.externalApi) {
+    return attachDefillamaStats(cmsSynth);
+  } else {
+    return fetchCompleteSynth(cmsSynth);
+  }
```

Comment:
If the `externalApi` field is filled in, then we use it instead of our api to fetch the `tvl` and `tvl24hChange ` values